sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -487,6 +508,9 @@ public void SetStartTime(DateTime startTime)
                 _currentActivity?.SetStartTime(startTime);
             }
 
+#if !NET5_0
+            [RequiresUnreferencedCode("The exception is used in a call to DiagnosticSource.Write, all necessary properties need to be preserved on the exception type being passed in using DynamicDependency attributes.")]
+#endif
             public void MarkFailed(Exception? exception)
             {
                 if (exception != null)
```

Comment:
Just a small improvement - if you feel like it:
As far as I can tell the `MarkFailed` is only called from this method (or rather, from internal methods, it's not a public API). So maybe we could use the same trick with generic parameters to avoid the `DynamicDependency` and move the solution for trimming one level down.

Something like:
```C#
[UnconditionalSuppressMessage("IL2026", .. same justification ...)]
public void MarkFailed<[DynamicallyAccessedMembers(PublicProperties)] T>(T exception) : where T : Exception
{
    ....
}
```

The advantage is that the caller of the MarkFailed doesn't need to do anything - it will just work there. Easier to maintain.

Reply:
You could then remove the `DynamicDependency` attributes around the `Exception` type...

Reply:
Updated!